src/mame/includes/contra.h

```diff
@@ -37,6 +37,11 @@ class contra_state : public driver_device
 		m_palette(*this, "palette")
 	{ }
 
+	uint32_t m_math_reg[6];
+	uint16_t m_multiply_result;
+	uint16_t m_divide_quotient;
+	uint16_t m_divide_remainder;
```

Comment:
The drives using this state class are flagged as supporting save states – you need to register stuff for saving.